Auth article completed, added a screenshot in resources

banani720 opened the pull request:
Feel free to recommend edits, add details, remove details, language errors etc.

Commit:
Auth article completed, added a screenshot in resources

sharifanani commented on content/post/2021-12-29-amplify.md:
```diff
@@ -0,0 +1,89 @@
+
+**Over the last couple of days,** I decided to take a bit of a dive into AWS's Amplify service and see what exactly it had to offer in terms of creating and deploying a web-based application. As Amazon describes it, "AWS Amplify is a set of purpose-built tools and features that lets frontend web and mobile developers quickly and easily build full-stack applications", so I was curious to see reall ...
+
+The project I decided to use this for used an Angular front-end. As for the backend services. The goal was to use as many of the services as possible provided in the Amplify ecosystem. Setting up amplify was simple enough. First, I needed to install the Amplify CLI via npm with `npm install -g @aws-amplify/cli` then run `amplify configure`, which provided a simple step-by-step guide to getting ho ...
```
The`As for backend services.` sentence feels misplaced, doesn't really read right. Was it meant to be there? Or is it supposed to read something like "As for backend services, the plan is to use only features offered by amplify"?

banani720 replied:
Fixed

sharifanani commented on content/post/2021-12-29-amplify.md:
```diff
@@ -0,0 +1,89 @@
+
+**Over the last couple of days,** I decided to take a bit of a dive into AWS's Amplify service and see what exactly it had to offer in terms of creating and deploying a web-based application. As Amazon describes it, "AWS Amplify is a set of purpose-built tools and features that lets frontend web and mobile developers quickly and easily build full-stack applications", so I was curious to see reall ...
+
+The project I decided to use this for used an Angular front-end. As for the backend services. The goal was to use as many of the services as possible provided in the Amplify ecosystem. Setting up amplify was simple enough. First, I needed to install the Amplify CLI via npm with `npm install -g @aws-amplify/cli` then run `amplify configure`, which provided a simple step-by-step guide to getting ho ...
```
For installation, show the installation for a specific version. This way your article isn't broken in the future. `npm install -g @aws-amplify/cli@<version-number>`

banani720 replied:
Added version number

sharifanani commented on content/post/2021-12-29-amplify.md:
```diff
@@ -0,0 +1,89 @@
+  - Amplify
+tags:
+  - Markdown
+  - 
```
```suggestion
  - AWS
  - Amplify
  - Angular
  - WebDev
  - Serverless
```

sharifanani commented on content/post/2021-12-29-amplify.md:
```diff
@@ -0,0 +1,89 @@
+
+Probably one of the most common features seen in web-apps today is authentication of some form or another. Be it for e-commerce, social media, or other web app categories, chances are it's there. To begin setting up the auth, the command `amplify add auth` brings up another step-by-step output in the terminal to configure the authentication processs, including configuration for 3rd party provider ...
+
+Implementing the authentication in the frontend can be done either manually, or through the use of template components. AWS provides a ready to use Authenticator component which can be popped into your application with a simple import. I decided to take a look at what they had for Angular, making use of their Amplify-UI guide which can be found [here](https://ui.docs.amplify.aws/getting-started/i ...
```
>The docs do a good job highlighting the module import you need to add to your `app.module.ts` file but it's what is unhighlighted and not included that are key to getting this component to work. Asides from the import of the `AmplifyAuthenticatorModule`, an import of `aws-config from ../aws-exports` and an import of `Amplify from 'aws-amplify'` are also required. You will also need to add below the import statements `Amplify.configure(awsconfig);`. This part in particular was completely ommitted from the documentation. Failing to perform these steps before serving up your application will lead to an error about Amplify not being configured properly.

I would immediately mention why they're required. Maybe phrase this more like:

>The docs do a good job highlighting the high-level imports you need, but leave little room for error. after _mention step from guide where you got stuck/stopped_, I ran into `actually copy and paste error here, people google their error messages all the time`. To get the component working, I needed to 
> ```
> import aws-config from '../aws-exports'
> import Amplify from 'aws-amplify'
> ```
>in `filename_here`

banani720 replied:
Added justification for imports and their importance

sharifanani commented on content/post/2021-12-29-amplify.md:
```diff
@@ -0,0 +1,89 @@
+
+Probably one of the most common features seen in web-apps today is authentication of some form or another. Be it for e-commerce, social media, or other web app categories, chances are it's there. To begin setting up the auth, the command `amplify add auth` brings up another step-by-step output in the terminal to configure the authentication processs, including configuration for 3rd party provider ...
+
+Implementing the authentication in the frontend can be done either manually, or through the use of template components. AWS provides a ready to use Authenticator component which can be popped into your application with a simple import. I decided to take a look at what they had for Angular, making use of their Amplify-UI guide which can be found [here](https://ui.docs.amplify.aws/getting-started/i ...
```
>`"allowJS":true`

Is this truly necessary? would a `// @ts-ignore` work for that import instead? Allowing JS throughout a project is usually a last resort I would think

banani720 replied:
Fixed

Commit:
Update content/post/2021-12-29-amplify.md

Co-authored-by: Sharif Anani <sharif.anani@gmail.com>

Commit:
Added version number, and changed period to comma

Added the version number for which version of Amplify was installed, also changed that backend period to a comma and uncapitalized the T in the

Commit:
Changed method from allowJS to @ts-ignore

Commit:
Added justification for why import aws-config and Amplify module

Commit:
regenerated site and removed old pages

Commit:
Updated dates, regenerated site
